uaclient/cli.py

```diff
@@ -477,6 +562,127 @@ def get_valid_entitlement_names(names: "List[str]"):
     return entitlements_found, entitlements_not_found
 
 
+def action_config(args, cfg, **kwargs):
+    """Perform the config action.
+
+    @return: 0 on success, 1 otherwise
+    """
+    parser = get_parser()
+    subparser = parser._get_positional_actions()[0].choices["config"]
+    valid_choices = subparser._get_positional_actions()[0].choices.keys()
+    if args.command not in valid_choices:
+        parser._get_positional_actions()[0].choices["config"].print_help()
+        raise exceptions.UserFacingError(
+            "\n<command> must be one of: {}".format(", ".join(valid_choices))
+        )
```

Comment:
Digging into the parser internals in this case makes me pause here.

Can we just assume that if this action got called, then there must not have been a sub-sub-command provided? and therefore we should always raise the UserFacingError (no need to check if the command is a valid choice)?

Or not, this is just a wishful suggestion

Reply:
I see now that the help message relies on having the `print_help()` and `valid_choices` which makes my suggestion not ideal, you can ignore

Reply:
Yep the print_help and valid_choices make for a much better UX. https://warthogs.atlassian.net/browse/SC-205 will address the use of parser private methods

Resolving